### ISSUE TITLE
Setting completing-read metadata to project-file a la `project-find-file`

### ISSUE THREAD

**Issue by zenspider** [redguardtoo/find-file-in-project#145]
While investigating problems with interactions between ffip and @protesilaos' [mct](https://github.com/protesilaos/mct) project, (in https://github.com/protesilaos/mct/issues/7), I've whipped up the following and it seems to work. Documentation for `ffip--file-completion-table` can be found in `21.6.7 Programmed Completion`. `completion-category-defaults` indicates that `project-file` is a regular category.

```elisp
(defun ffip--file-completion-table (all-files)
  ;; direct copy of project--file-completion-table
  (lambda (string pred action)
    (cond
     ((eq action 'metadata)
      '(metadata . ((category . project-file))))
     (t
      (complete-with-action action all-files string pred)))))

(defun ffip-completing-read (prompt collection &optional action)
  "Read a string in minibuffer, with completion.

PROMPT is a string with same format parameters in `completing-read'.
COLLECTION is a list of strings.

ACTION is a lambda function to call after selecting a result.

This function returns the selected candidate or nil."
  (let* (selected)
    (cond
     ((= 1 (length collection))
      ;; select the only candidate immediately
      (setq selected (car collection)))

     (ffip-prefer-ido-mode
      ;; ido can only handle list of strings
      (setq selected (ido-completing-read prompt (mapcar 'car collection))))

     (t
      (setq selected (completing-read prompt (ffip--file-completion-table collection)))
      (setq selected (or (assoc selected collection) selected))))

    (when selected
      ;; make sure only the string/file is passed to action
      (let* ((default-directory (ffip-get-project-root-directory))
             (result (if (consp selected) (cdr selected) selected)))
        (if action (funcall action result) result)))))
```

Possibly related to #144 but they didn't provide enough info to tell... but `'file` isn't a category listed in `completion-category-defaults` so I suspect it is something else.

**Comment by zenspider** [comment]
I'm happy to convert this issue into a real PR if you're interested.

**Comment by redguardtoo** [comment]
What's mct? Could you give me more details? 

PR is welcomed if existing functionalities are not broken. Please do a full regression test before submitting PR.

**Comment by zenspider** [comment]
I've referenced all the information you've asked about.

**Comment by zenspider** [comment]
Closed by #147
